src/configuration/mod.rs

```diff
@@ -1,56 +1,339 @@
 //! Configuration module
 
-mod prima;
-mod test;
+use crate::error::Error as PrimaDatadogError;
+use crate::TagTrackerConfiguration;
+use std::fmt::Display;
+use std::str::FromStr;
 
-pub use prima::{Config, Country, Environment};
-pub use test::TestConfiguration;
+/// By binding to 0.0.0.0:0 we're just letting the OS assign us a port, and letting anyone send us UDP packets on that port
+const DEFAULT_FROM_ADDR: &str = "0.0.0.0:0";
 
-use crate::TagTrackerConfiguration;
+/// The struct that represents options for the Datadog client in Prima.
+pub struct Configuration {
```

Comment:
Do we want to impl Deserialize here with the serde feature?

Reply:
Don't know, not sure what we do with serde